mlrun/projects/operations.py

```diff
@@ -262,6 +263,16 @@ def build_function(
     :param overwrite_build_params:  overwrite the function build parameters with the provided ones, or attempt to add
      to existing parameters
     """
+
+    def _remove_image_protocol_prefix(image_name):
+        prefixes = ["https://", "https://"]
+        if any(prefix in image_name for prefix in prefixes):
+            warnings.warn(
+                "The image has an unexpected protocol prefix, The prefix was removed."
+            )
+            image_name = image_name.removeprefix("https://").removeprefix("http://")
+        return image_name
+
```

Comment:
this doesn't covers users who are using `function.deploy()`

Reply:
what about handling it on API side? to cover use cases when deploying from older clients or from UI.